diff --git a/src/plugins/multimedia/ffmpeg/qffmpegmediaintegration.cpp b/src/plugins/multimedia/ffmpeg/qffmpegmediaintegration.cpp
index ba1fff3b3..18e1d3599 100644
--- a/src/plugins/multimedia/ffmpeg/qffmpegmediaintegration.cpp
+++ b/src/plugins/multimedia/ffmpeg/qffmpegmediaintegration.cpp
@@ -356,15 +356,12 @@ Q_DECL_EXPORT jint JNICALL JNI_OnLoad(JavaVM *vm, void * /*reserved*/)
 
     QT_USE_NAMESPACE
     void *environment;
-    if (vm->GetEnv(&environment, JNI_VERSION_1_6))
+    if (vm->GetEnv(&environment, JNI_VERSION_1_6) != JNI_OK)
         return JNI_ERR;
 
-    // setting our javavm into ffmpeg.
-    if (av_jni_set_java_vm(vm, nullptr))
-        return JNI_ERR;
+    av_jni_set_java_vm(vm, nullptr);
 
-    if (!QAndroidCamera::registerNativeMethods())
-        return JNI_ERR;
+    QAndroidCamera::registerNativeMethods();
 
     return JNI_VERSION_1_6;
 }
