diff --git a/src/linguist/CMakeLists.txt b/src/linguist/CMakeLists.txt
index 1686f470e..c4bc7a201 100644
--- a/src/linguist/CMakeLists.txt
+++ b/src/linguist/CMakeLists.txt
@@ -5,16 +5,13 @@ qt_exclude_tool_directories_from_default_target(
     linguist
 )
 
-if(NOT QT_FEATURE_linguist)
-    return()
-endif()
 add_subdirectory(lconvert)
 add_subdirectory(lprodump)
 add_subdirectory(lrelease)
 add_subdirectory(lrelease-pro)
 add_subdirectory(lupdate)
 add_subdirectory(lupdate-pro)
-if(QT_FEATURE_process AND QT_FEATURE_pushbutton AND QT_FEATURE_toolbutton
+if(QT_FEATURE_linguist AND QT_FEATURE_process AND QT_FEATURE_pushbutton AND QT_FEATURE_toolbutton
     AND QT_FEATURE_png AND QT_FEATURE_mdiarea AND QT_FEATURE_syntaxhighlighter
     AND TARGET Qt::Widgets)
     add_subdirectory(linguist)
