diff --git a/src/linguist/CMakeLists.txt b/src/linguist/CMakeLists.txt
index 195e4bb4a..9b2e3e617 100644
--- a/src/linguist/CMakeLists.txt
+++ b/src/linguist/CMakeLists.txt
@@ -5,10 +5,6 @@ qt_exclude_tool_directories_from_default_target(
     linguist
 )
 
-if(NOT QT_FEATURE_linguist)
-    return()
-endif()
-
 add_subdirectory(shared/qmake-parser)
 add_subdirectory(shared/trlib)
 add_subdirectory(shared/linguistproject)
@@ -20,7 +16,7 @@ add_subdirectory(lrelease-pro)
 add_subdirectory(lupdate)
 add_subdirectory(lupdate-pro)
 add_subdirectory(ltext2id)
-if(QT_FEATURE_process AND QT_FEATURE_pushbutton AND QT_FEATURE_toolbutton
+if(QT_FEATURE_linguist AND QT_FEATURE_process AND QT_FEATURE_pushbutton AND QT_FEATURE_toolbutton
     AND QT_FEATURE_png AND QT_FEATURE_mdiarea AND QT_FEATURE_syntaxhighlighter
     AND TARGET Qt::Widgets AND TARGET Qt::Quick)
     add_subdirectory(linguist)
