diff --git a/cmake/FindWrapOpenSSLHeaders.cmake b/cmake/FindWrapOpenSSLHeaders.cmake
index 510ae10854..e534857aab 100644
--- a/cmake/FindWrapOpenSSLHeaders.cmake
+++ b/cmake/FindWrapOpenSSLHeaders.cmake
@@ -30,7 +30,11 @@ if(OPENSSL_INCLUDE_DIR)
 
     add_library(WrapOpenSSLHeaders::WrapOpenSSLHeaders INTERFACE IMPORTED)
     target_include_directories(WrapOpenSSLHeaders::WrapOpenSSLHeaders INTERFACE
-        ${OPENSSL_INCLUDE_DIR})
+        "${OPENSSL_ROOT_DIR}/${OPENSSL_INCLUDE_DIR}")
+    set_target_properties(OpenSSL::SSL PROPERTIES
+        INTERFACE_INCLUDE_DIRECTORIES "${OPENSSL_ROOT_DIR}/${OPENSSL_INCLUDE_DIR}")
+    set_target_properties(OpenSSL::Crypto PROPERTIES
+        INTERFACE_INCLUDE_DIRECTORIES "${OPENSSL_ROOT_DIR}/${OPENSSL_INCLUDE_DIR}")
 
     set_target_properties(WrapOpenSSLHeaders::WrapOpenSSLHeaders PROPERTIES
         _qt_is_nolink_target TRUE)
