diff --git a/src/corelib/Qt6CoreMacros.cmake b/src/corelib/Qt6CoreMacros.cmake
index 18576eead0..a45d6870fc 100644
--- a/src/corelib/Qt6CoreMacros.cmake
+++ b/src/corelib/Qt6CoreMacros.cmake
@@ -2519,7 +2519,7 @@ function(_qt_internal_setup_deploy_support)
         set(safe_target_file
             "$<TARGET_FILE:$<IF:${have_deploy_tool},${target_if_exists},${target}>>")
         set(__QT_DEPLOY_TOOL "$<IF:${have_deploy_tool},${safe_target_file},${fallback}>")
-    elseif(UNIX AND NOT APPLE AND NOT ANDROID AND NOT CMAKE_CROSSCOMPILING)
+    elseif(UNIX AND NOT APPLE AND NOT ANDROID)
         set(__QT_DEPLOY_TOOL "GRD")
     else()
         # Android is handled as a build target, not via this install-based approach.
