diff --git a/qtdeclarative/src/qml/qml/qqmldirparser.cpp b/qtdeclarative/src/qml/qml/qqmldirparser.cpp
index d87bf433b8..eb9303fed7 100644
--- a/qtdeclarative/src/qml/qml/qqmldirparser.cpp
+++ b/qtdeclarative/src/qml/qml/qqmldirparser.cpp
@@ -106,6 +106,7 @@ bool QQmlDirParser::parse(const QString &source)
     _plugins.clear();
     _components.clear();
     _scripts.clear();
+	_typeNamespace="";
     _designerSupported = false;
     _className.clear();
 
